kratos/geometries/geometry_shape_function_container.h

```diff
@@ -0,0 +1,273 @@
+    {
+        return mShapeFunctionsValues[ThisMethod];
+    }
+
+    double ShapeFunctionValue(
+        IndexType IntegrationPointIndex,
+        IndexType ShapeFunctionIndex,
+        IntegrationMethod ThisMethod ) const
+    {
+        KRATOS_DEBUG_ERROR_IF(mShapeFunctionsValues[ThisMethod].size1() <= IntegrationPointIndex )
+            << "No existing integration point" << std::endl;
+
+        KRATOS_DEBUG_ERROR_IF(mShapeFunctionsValues[ThisMethod].size2() <= ShapeFunctionIndex )
+            << "No existing shape function value" << std::endl;
+
+        return mShapeFunctionsValues[ThisMethod]( IntegrationPointIndex, ShapeFunctionIndex );
+    }
+
+    const ShapeFunctionsGradientsType& ShapeFunctionsLocalGradients(
+        IntegrationMethod ThisMethod ) const
+    {
+        return mShapeFunctionsLocalGradients[ThisMethod];
+    }
+
+    const Matrix& ShapeFunctionLocalGradient(
+        IndexType IntegrationPointIndex,
+        IntegrationMethod ThisMethod ) const
+    {
+        KRATOS_DEBUG_ERROR_IF(mShapeFunctionsLocalGradients[ThisMethod].size() <= IntegrationPointIndex )
+            << "No existing integration point" << std::endl;
+
+        return mShapeFunctionsLocalGradients[ThisMethod][IntegrationPointIndex];
+    }
+
+    const Matrix& ShapeFunctionLocalGradient(
+        IndexType IntegrationPointIndex,
+        IndexType ShapeFunctionIndex,
+        IntegrationMethod ThisMethod ) const
+    {
+        KRATOS_DEBUG_ERROR_IF(mShapeFunctionsLocalGradients[ThisMethod].size() <= IntegrationPointIndex )
+            << "No existing integration point" << std::endl;
+
+        return mShapeFunctionsLocalGradients[ThisMethod][IntegrationPointIndex];
+    }
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /** Turn back information as a string.
+
+    @return String contains information about this geometry.
+    @see PrintData()
+    @see PrintInfo()
+    */
+    virtual std::string Info() const
+    {
+        return "geometry data";
+    }
+
+    /** Print information about this object.
+
+    @param rOStream Stream to print into it.
+    @see PrintData()
+    @see Info()
+    */
+    virtual void PrintInfo( std::ostream& rOStream ) const
+    {
+        rOStream << "shape function container";
+    }
+
+    /** Print geometry's data into given stream. Prints it's points
+    by the order they stored in the geometry and then center
+    point of geometry.
+
+    @param rOStream Stream to print into it.
+    @see PrintInfo()
+    @see Info()
+    */
+    virtual void PrintData( std::ostream& rOStream ) const
+    {
+    }
+
+    ///@}
+
+protected:
+
+private:
+    ///@name Member Variables
+    ///@{
+
+    ShapeFunctionsValuesContainerType mShapeFunctionsValues;
+
+    ShapeFunctionsLocalGradientsContainerType mShapeFunctionsLocalGradients;
+
+    ///@}
+    ///@name Serialization
+    ///@{
+
+    friend class Serializer;
+
+    virtual void save( Serializer& rSerializer ) const
+    {
+    }
```

Comment:
I understand that now all the shape functions are static. But for Breps you need to save and load the values. Isn't it?